packages/esm-appointments-app/src/form/appointments-form.resource.ts

```diff
@@ -0,0 +1,126 @@
+import dayjs from 'dayjs';
+import useSWR from 'swr';
+import { openmrsFetch } from '@openmrs/esm-framework';
+import {
+  type AppointmentPayload,
+  type AppointmentService,
+  type AppointmentsFetchResponse,
+  type RecurringAppointmentsPayload,
+} from '../types';
+import isToday from 'dayjs/plugin/isToday';
+dayjs.extend(isToday);
+
+const appointmentsSearchUrl = `/ws/rest/v1/appointments/search`;
+
+export function useAppointments(patientUuid: string, startDate: string, abortController: AbortController) {
+  /*
+    SWR isn't meant to make POST requests for data fetching. This is a consequence of the API only exposing this resource via POST.
+    This works but likely isn't recommended.
+  */
+  const fetcher = () =>
+    openmrsFetch(appointmentsSearchUrl, {
+      method: 'POST',
+      signal: abortController.signal,
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: {
+        patientUuid: patientUuid,
+        startDate: startDate,
+      },
+    });
+
+  const { data, error, isLoading, isValidating, mutate } = useSWR<AppointmentsFetchResponse, Error>(
+    appointmentsSearchUrl,
+    fetcher,
+  );
+
+  const appointments = data?.data?.length ? data.data : null;
+
+  const pastAppointments = appointments
+    ?.sort((a, b) => (b.startDateTime > a.startDateTime ? 1 : -1))
+    ?.filter(({ status }) => status !== 'Cancelled')
+    ?.filter(({ startDateTime }) =>
+      dayjs(new Date(startDateTime).toISOString()).isBefore(new Date().setHours(0, 0, 0, 0)),
+    );
+
+  const upcomingAppointments = appointments
+    ?.sort((a, b) => (a.startDateTime > b.startDateTime ? 1 : -1))
+    ?.filter(({ status }) => status !== 'Cancelled')
+    ?.filter(({ startDateTime }) => dayjs(new Date(startDateTime).toISOString()).isAfter(new Date()));
+
+  const todaysAppointments = appointments
+    ?.sort((a, b) => (a.startDateTime > b.startDateTime ? 1 : -1))
+    ?.filter(({ status }) => status !== 'Cancelled')
+    ?.filter(({ startDateTime }) => dayjs(new Date(startDateTime).toISOString()).isToday());
```

Comment:
This seems refactorable:

```ts
const activeAppointments = appointments
  ?.filter(({ status }) => status !== 'Cancelled') ?? [];

// no ? here since `activeAppointments` is guaranteed to be an array
const pastAppointments = activeAppointments
  .sort((a, b) => (b.startDateTime > a.startDateTime ? 1 : -1))
  .filter(({ startDateTime }) =>
      dayjs(new Date(startDateTime).toISOString()).isBefore(new Date().setHours(0, 0, 0, 0)),
    );
```

Not essential.

Reply:
Will take a look, just copied this code over from patient chart so hesitant to refactor unless I can clearly test.